### PR TITLE
add /api/backup in monitor to backup boltdb

### PR DESCRIPTION
**Pull request by therealssj**
--- Old ---
- exposed `db` to teller 
- create a `Backup` function in teller which returns `http.Handlerfunc` to start backup download
- created a `Backuper` interface in `monitor` to expose the `Backup` function from `teller` to `monitor`

--- Edit ---
- expose `db` to monitor 
- create a `Backup` function in monitor which returns `http.Handlerfunc` to start backup download

### CODE REVIEW

**Comment by therealssj** [comment]
if this is fine. can you checkout and test before merge?

File: src/teller/teller.go [hunk 1]
```diff
@@ -71,6 +78,21 @@ func (s *Teller) Shutdown() {
 	<-s.done
 }
 
+func (s *Teller) Backup() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		err := s.db.View(func(tx *bolt.Tx) error {
+			w.Header().Set("Content-Type", "application/octet-stream")
+			w.Header().Set("Content-Disposition", `attachment; filename="teller.db"`)
```
Comment (gz-c) [inline src/teller/teller.go]: timestamp the filename

File: src/teller/teller.go [hunk 2]
```diff
@@ -71,6 +78,21 @@ func (s *Teller) Shutdown() {
...
+			w.Header().Set("Content-Length", strconv.Itoa(int(tx.Size())))
+			_, err := tx.WriteTo(w)
+			return err
+		})
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
```
Comment (gz-c) [inline src/teller/teller.go]: log if this happens

**Review by gz-c — CHANGES_REQUESTED** [review]

**Comment by gz-c** [comment]
update readme

**Comment by therealssj** [comment]
I will update readme after your monitor PR merge

**Comment by gz-c** [comment]
merged monitor PR

**Comment by therealssj** [comment]
Done.

File: src/teller/teller.go [hunk 3]
```diff
@@ -71,6 +81,24 @@ func (s *Teller) Shutdown() {
 	<-s.done
 }
 
+// Backup starts a timestamped backup download
+func (s *Teller) Backup() http.HandlerFunc {
```
Comment (gz-c) [inline src/teller/teller.go]: Why not pass the db to Monitor and add the handler there?

**Review by gz-c — CHANGES_REQUESTED** [review]

File: src/teller/teller.go (hunk 3 above)
Comment (therealssj) [reply src/teller/teller.go]: More semantic?
Like saying taking backup for teller

Comment (gz-c) [reply src/teller/teller.go]: seems like indirection, only monitor would ever use it, 

teller.Teller exposes the public bind/status API, backup is not a function of the client interface

**Comment by therealssj** [comment]
@gz-c refactored

**Pull request merged**